collector/collector.go

```diff
@@ -119,6 +119,16 @@ func NewNodeCollector(logger log.Logger, filters ...string) (*NodeCollector, err
 	collectors := make(map[string]Collector)
 	initiatedCollectorsMtx.Lock()
 	defer initiatedCollectorsMtx.Unlock()
+
+	// The meminfo collector is being deprecated in favor of the new
+	// meminfo_procfs collector. They are mutually exclusive and should not
+	// be enabled at the time same.
+	micEnabled, micExist := collectorState["meminfo"]
+	mipcEnabled, mipcExist := collectorState["meminfo_procfs"]
+	if (micExist && mipcExist) && (*micEnabled && *mipcEnabled) {
+		return nil, fmt.Errorf("'meminfo' and 'meminfo_procfs' collectors are mutually exclusive, please disable one")
+	}
+
```

Comment:
Currently, this results in a panic:

```bash
~/go/src/github.com/prometheus/node_exporter (feat/procfslib-memory-collector [ U ]) -> ./node_exporter --collector.meminfo_procfs --log.level="debug" --web.listen-address ":9101"
ts=2024-06-08T18:12:51.917Z caller=node_exporter.go:193 level=info msg="Starting node_exporter" version="(version=1.8.1, branch=feat/procfslib-memory-collector, revision=1bab3570959f98b7e5b833d94c7024c7658c63aa)"
ts=2024-06-08T18:12:51.917Z caller=node_exporter.go:194 level=info msg="Build context" build_context="(go=go1.22.3, platform=linux/amd64, user=tjhop@contraband, date=20240608-18:05:10, tags=unknown)"
ts=2024-06-08T18:12:51.917Z caller=node_exporter.go:199 level=debug msg="Go MAXPROCS" procs=1
panic: Couldn't create metrics handler: couldn't create collector: 'meminfo' and 'meminfo_procfs' collectors are mutually exclusive, please disable one

goroutine 1 [running]:
main.newHandler(0x1, 0x28, {0xcf2740, 0xc0000d0440})
        /home/tjhop/go/src/github.com/prometheus/node_exporter/node_exporter.go:69 +0x2ab
main.main()
        /home/tjhop/go/src/github.com/prometheus/node_exporter/node_exporter.go:201 +0x128d
```

This definitely achieves the goal of "don't run the node exporter if both flags are enabled", but is certainly not a graceful exit, either. 

Is there a better place to check and ensure both collectors aren't enabled at the same time? I had considered doing so in `registerCollector()`, but it felt awkward trying to fatal out from there. I had also considered checking in the `init()` funcs of both collectors when they try to do their self registration to only register if $theOther isn't registered yet, but I wasn't sure about any ordering that may/may not exist in collector registration and didn't want to cause a race between the 2 to see which would successfully register.